### PR TITLE
create PatchSIngle endpoint, create Integration Tests

### PR DESCRIPTION
**Pull request by badekk**
## create PatchSIngle endpoint, create Integration Tests

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How has this been tested?

- [ ] Unit test(s)
- [x] Integration test(s)
- [ ] Tested manually

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I have checked my code and corrected any misspellings
- [ ] I have added frontend code and checked it for mobile & desktop

### CODE REVIEW

**Comment by mar-pan** [comment]
Task linked: [CU-862j9cr7u Patch single](https://app.clickup.com/t/862j9cr7u)

File: src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            {
+                _logger.LogError("Couldn't find user by id in database");
+                throw new ItemNotFoundException("Sorry we are unable to find this user in database");
+            }
+
+            if (!payload.Roles.Any())
+            {
+                _logger.LogError("There were no roles in the payload");
+                throw new ClientInputErrorException("Sorry we couldn't fetch roles");
+            }
```
Comment (h4570) [inline src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs:59]: No czemu, przecież rozmawialiśmy o tym. 
Jeśli ktoś nie będzie miał ról w bazce, to przy próbie naprawy poprzez PATCH, kod będzie zawsze tu wpadać i nie będziemy w stanie tego naprawić.

File: src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            var roleNamesToAdd = payload.Roles.Where(c => !dbResult.Roles!.Select(c => c.Role!.Name).Contains(Enum.Parse<RoleName>(c)));
+            var rolesToRemove = dbResult.Roles!.Where(c => !payload.Roles.Contains(c.Role!.Name.ToString())).ToList();
+
+            foreach (var role in rolesToRemove)
+            {
+                dbResult.Roles!.Remove(role);
+            }
+
+            var rolesToAdd = await _appDbContext.Roles
+                .ToListAsync(); // materialize the query
```
Comment (h4570) [inline src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs:70]: kom do wywalenia

File: src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+            rolesToAdd = rolesToAdd
+                .Where(r => roleNamesToAdd.Contains(r.Name.ToString())) // perform the Contains locally
```
Comment (h4570) [inline src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs:73]: kom do wywalenia

File: src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                .ToList();
+            foreach (var role in rolesToAdd)
```
Comment (h4570) [inline src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs:75]: enter przed `foreach`

File: src/backend/AuthTests/Actions/ManageUsersActionsTest/PatchSingleTest/PatchSingleActionTest.cs [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+            var testUser1Role = PatchSingleActionFixture.CreateUserRole(testingUser1, roleDonorId);
+            var testUser1Role2 = PatchSingleActionFixture.CreateUserRole(testingUser1, roleAdminId);
+            var testUser1Role3 = PatchSingleActionFixture.CreateUserRole(testingUser1, roleDelivererId);
+
+            context.Add(testingUser1);
+            context.Add(testingAuth0User1);
+            context.Add(testUser1Role);
+            context.Add(testUser1Role2);
+            context.Add(testUser1Role3);
+
```
Comment (h4570) [inline src/backend/AuthTests/Actions/ManageUsersActionsTest/PatchSingleTest/PatchSingleActionTest.cs:52]: elegancko

**Review by h4570 — CHANGES_REQUESTED** [review]

File: src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs (hunk 1 above)
Comment (badekk) [reply src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs:59]: tzn. zmieniłem z bazy na payload, żeby w momencie gdy chcesz wykonac patcha, ale w payloadzie nie ma ról żadnych, to wtedy ma bład wyrzucić, ale w sumie to nie będzie można Patch'em usunąć ról. Tylko nadać conajmniej 1.

Comment (h4570) [reply src/backend/Auth/Actions/ManageUsersActions/ManagerUsersPatchSingle/ManageUsersPatchSingleService.cs:59]: Sorry mój blad

**Review by h4570 — APPROVED** [review]

**Pull request merged**